h3 index for center point is not returned during polyfill

dear h3 developers.


When polyfilling the radius for a specific point, there is an issue where the h3 cell is not filled for the center point as shown below. 

## Case1. Using JavaScript / Resolution 10
When I tested the [h3geo.org](https://h3geo.org/docs/api/regions) polyfill function, it caused the same issue.
**JavaScript (Live) - LIVE EDITOR**

```
function example() {
    const polygon = [
        [51.51746913670682, -0.1251052178242358],
        [51.51981162007689, -0.1254345584085245],
        [51.522082928275104, -0.1264125733143912],
        [51.52421404875121, -0.1280095460538208],
        [51.52614022848794, -0.1301769534182327],
        [51.52780294148869, -0.1328489398316205],
        [51.52915166706029, -0.1359443183389036],
        [51.53014542485875, -0.1393690374304754],
        [51.53075402005622, -0.1430190387497298],
        [51.5309589607956, -0.1467834188535714],
        [51.53075402005622, -0.1505477989574129],
        [51.53014542485875, -0.1541978002766674],
        [51.52915166706029, -0.1576225193682391],
        [51.52780294148869, -0.1607178978755222],
        [51.52614022848794, -0.16338988428891],
        [51.52421404875121, -0.1655572916533219],
        [51.522082928275104, -0.1671542643927515],
        [51.51981162007689, -0.1681322792986182],
        [51.51746913670682, -0.1684616198829069],
        [51.51512665333676, -0.1681322792986182],
        [51.51285534513854, -0.1671542643927515],
        [51.51072422466243, -0.1655572916533219],
        [51.508798044925705, -0.16338988428891],
        [51.50713533192496, -0.1607178978755222],
        [51.505786606353354, -0.1576225193682392],
        [51.50479284855489, -0.1541978002766674],
        [51.50418425335742, -0.1505477989574129],
        [51.50397931261804, -0.1467834188535714],
        [51.50418425335742, -0.1430190387497299],
        [51.50479284855489, -0.1393690374304753],
        [51.505786606353354, -0.1359443183389036],
        [51.50713533192496, -0.1328489398316205],
        [51.508798044925705, -0.1301769534182327],
        [51.51072422466243, -0.1280095460538208],
        [51.51285534513854, -0.1264125733143912],
        [51.51512665333676, -0.1254345584085245],
        [51.51746913670682, -0.1251052178242358]
    ];
    const res = 10;
    return h3.polyfill(polygon, res);
}
```

**JavaScript (Live) - RESULT**
```
["8a195da49757fff","8a195da48a0ffff","8a195da4d207fff","8a195da48257fff","8a195da49ce7fff","8a195da4d797fff","8a195da4952ffff","8a195da4802ffff","8a195da48b4ffff","8a195da49897fff","8a195da48397fff","8a195da490dffff","8a195da4816ffff","8a195da49447fff","8a195da4d6affff","8a195da4d487fff","8a195da497affff","8a195da48a67fff","8a195da482affff","8a195da49587fff","8a195da49b17fff","8a195da4bb37fff","8a195da48ba7fff","8a195da498effff","8a195da4d39ffff","8a195da4b90ffff","8a195da49137fff","8a195da49c57fff","8a195da4d707fff","8a195da4949ffff","8a195da4d4dffff","8a195da49807fff","8a195da4b827fff","8a195da4904ffff","8a195da4d2b7fff","8a195da48307fff","8a195da49d97fff","8a195da4d08ffff","8a195da4866ffff","8a195da493b7fff","8a195da4d61ffff","8a195da49947fff","8a195da4b967fff","8a195da4918ffff","8a195da4971ffff","8a195da4821ffff","8a195da49caffff","8a195da494f7fff","8a195da4d75ffff","8a195da49a87fff","8a195da48b17fff","8a195da4985ffff","8a195da490a7fff","8a195da4835ffff","8a195da49637fff","8a195da4d0e7fff","8a195da4940ffff","8a195da4d677fff","8a195da4999ffff","8a195da4d44ffff","8a195da49777fff","8a195da48a2ffff","8a195da48277fff","8a195da49d07fff","8a195da4d7b7fff","8a195da4954ffff","8a195da4804ffff","8a195da48b6ffff","8a195da498b7fff","8a195da483b7fff","8a195da49c1ffff","8a195da49467fff","8a195da4d6cffff","8a195da48a87fff","8a195da482cffff","8a195da49017fff","8a195da49d5ffff","8a195da4885ffff","8a195da495a7fff","8a195da49b37fff","8a195da4990ffff","8a195da4b92ffff","8a195da49157fff","8a195da49c77fff","8a195da4d727fff","8a195da49827fff","8a195da48adffff","8a195da4b847fff","8a195da4906ffff","8a195da4d2d7fff","8a195da48327fff","8a195da49db7fff","8a195da49b8ffff","8a195da49967fff","8a195da491affff","8a195da49ccffff","8a195da49517fff","8a195da49aa7fff","8a195da48b37fff","8a195da490c7fff","8a195da48157fff","8a194ad32597fff","8a195da4942ffff","8a195da4d697fff","8a195da4d46ffff","8a195da49797fff","8a195da48a4ffff","8a195da48297fff","8a195da49d27fff","8a195da4956ffff","8a195da4806ffff","8a195da4bb1ffff","8a195da48b8ffff","8a195da498d7fff","8a195da4d387fff","8a195da4911ffff","8a195da49487fff","8a195da4d6effff","8a195da49a17fff","8a195da4d4c7fff","8a195da48aa7fff","8a195da4b80ffff","8a195da4d29ffff","8a195da482effff","8a195da49037fff","8a195da480c7fff","8a195da4bb77fff","8a195da4d607fff","8a195da4992ffff","8a195da4b94ffff","8a195da49177fff","8a195da49707fff","8a195da48207fff","8a195da49c97fff","8a195da494dffff","8a195da4d747fff","8a195da49847fff","8a195da4d2f7fff","8a195da4b867fff","8a195da4908ffff","8a195da48347fff","8a195da49baffff","8a195da4d0cffff","8a195da4d65ffff","8a195da49987fff","8a195da48a17fff","8a195da4d20ffff","8a195da4825ffff","8a195da49ceffff","8a195da4d79ffff","8a195da49537fff","8a195da48b57fff","8a195da4989ffff","8a195da4839ffff","8a195da490e7fff","8a194ad325b7fff","8a195da4944ffff","8a195da4d6b7fff","8a195da4d48ffff","8a195da497b7fff","8a195da48a6ffff","8a195da482b7fff","8a195da49d47fff","8a195da4958ffff","8a195da49b1ffff","8a195da48baffff","8a195da498f7fff","8a195da4d3a7fff","8a195da4b917fff","8a195da49c5ffff","8a195da4d70ffff","8a195da494a7fff","8a195da49a37fff","8a195da4d4e7fff","8a195da4980ffff","8a195da48ac7fff","8a195da4b82ffff","8a195da49057fff","8a195da4830ffff","8a195da49d9ffff","8a195da4d097fff","8a195da480e7fff","8a195da4d627fff","8a195da4994ffff","8a195da4b96ffff","8a195da49197fff","8a195da49727fff","8a195da48227fff","8a195da49cb7fff","8a195da4d767fff","8a195da49867fff","8a195da48b1ffff","8a195da490affff","8a195da48367fff","8a195da4d0effff","8a195da49417fff","8a195da499a7fff","8a195da4d457fff","8a195da48a37fff","8a195da49d0ffff","8a195da49557fff","8a195da48057fff","8a194ad32497fff","8a195da4bb07fff","8a195da48b77fff","8a195da49107fff","8a195da49c27fff","8a195da4946ffff","8a195da4d6d7fff","8a195da4d4affff","8a195da48a8ffff","8a195da4d287fff","8a195da482d7fff","8a195da4901ffff","8a195da49d67fff","8a195da495affff","8a195da49917fff","8a195da4b937fff","8a195da4915ffff","8a195da4d72ffff","8a195da494c7fff","8a195da4982ffff","8a195da48ae7fff","8a195da4b84ffff","8a195da4d2dffff","8a195da49077fff","8a195da4832ffff","8a195da49b97fff","8a195da4d647fff","8a195da4996ffff","8a195da4d41ffff","8a195da491b7fff","8a195da49747fff","8a195da48247fff","8a195da49cd7fff","8a195da4d787fff","8a195da4951ffff","8a195da49aaffff","8a195da49887fff","8a195da490cffff","8a195da48387fff","8a195da4815ffff","8a194ad3259ffff","8a195da49437fff","8a195da4d69ffff","8a195da4d477fff","8a195da4979ffff","8a195da48a57fff","8a195da4829ffff","8a195da49d2ffff","8a195da49577fff","8a195da48077fff","8a195da49b07fff","8a194ad324b7fff","8a195da4bb27fff","8a195da48b97fff","8a195da498dffff","8a195da4d38ffff","8a195da49127fff","8a195da49c47fff","8a195da48747fff","8a195da4948ffff","8a195da4d6f7fff","8a195da4d4cffff","8a195da48aaffff","8a195da4d2a7fff","8a195da482f7fff","8a195da49d87fff","8a195da480cffff","8a195da493a7fff","8a195da4d60ffff","8a195da49937fff","8a195da4b957fff","8a195da4970ffff","8a195da4820ffff","8a195da49c9ffff","8a195da4d74ffff","8a195da494e7fff","8a195da48b07fff","8a195da4984ffff","8a195da4b86ffff","8a195da49097fff","8a195da4834ffff","8a195da49627fff","8a195da4d0d7fff","8a195da49bb7fff","8a195da4d667fff","8a195da4998ffff","8a195da4b9affff","8a195da49767fff","8a195da48a1ffff","8a195da4d217fff","8a195da48267fff","8a195da49cf7fff","8a195da4d7a7fff","8a195da48b5ffff","8a195da498a7fff","8a195da490effff","8a195da483a7fff","8a195da49c0ffff","8a195da49457fff","8a195da48a77fff","8a195da49007fff","8a195da49d4ffff","8a195da4884ffff","8a195da49597fff","8a195da4bb47fff","8a195da48bb7fff","8a195da4d3affff","8a195da4b91ffff","8a195da49147fff","8a195da49c67fff","8a195da48767fff","8a195da4d717fff","8a195da494affff","8a195da4d4effff","8a195da49817fff","8a195da48acffff","8a195da4b837fff","8a195da4d2c7fff","8a195da4905ffff","8a195da48317fff","8a195da49da7fff","8a195da4d09ffff","8a195da480effff","8a195da4d62ffff","8a195da49957fff","8a195da4b977fff","8a195da4919ffff","8a195da4972ffff","8a195da4822ffff","8a195da4d76ffff","8a195da49507fff","8a195da49a97fff","8a195da48b27fff","8a195da4986ffff","8a195da490b7fff","8a195da4836ffff","8a195da4d0f7fff","8a195da48147fff","8a194ad32587fff","8a195da4941ffff","8a195da4d687fff","8a195da499affff","8a195da4d45ffff","8a195da49787fff","8a195da4d237fff","8a195da48287fff","8a195da49d17fff","8a195da4955ffff","8a194ad3249ffff","8a195da4805ffff","8a195da4bb0ffff","8a195da49337fff","8a195da498c7fff","8a195da4910ffff","8a195da49c2ffff","8a195da49477fff","8a195da4d6dffff","8a195da48a97fff","8a195da4d28ffff","8a195da482dffff","8a195da49027fff","8a195da49d6ffff","8a195da4886ffff","8a195da495b7fff","8a195da48647fff","8a195da4bb67fff","8a195da4991ffff","8a195da49167fff","8a195da49c87fff","8a195da4d737fff","8a195da494cffff","8a195da49837fff","8a195da48aeffff","8a195da4d2e7fff","8a195da4b857fff","8a195da48337fff","8a195da49b9ffff","8a195da4d64ffff","8a195da49977fff","8a195da48a07fff","8a195da4824ffff","8a195da49cdffff","8a195da4d78ffff","8a195da49527fff","8a195da49ab7fff","8a195da48b47fff","8a195da4988ffff","8a195da4838ffff","8a195da490d7fff","8a195da48167fff","8a195da4d6a7fff","8a195da497a7fff","8a195da48a5ffff","8a195da4d257fff","8a195da482a7fff","8a195da49d37fff","8a195da4bb2ffff","8a195da48b9ffff","8a195da498e7fff","8a195da4d397fff","8a195da4b907fff","8a195da4912ffff","8a195da49c4ffff","8a195da4874ffff","8a195da49497fff","8a195da4d4d7fff","8a195da48ab7fff","8a195da4d2affff","8a195da49047fff","8a195da49d8ffff","8a195da4d087fff","8a195da48667fff","8a195da4d617fff","8a195da4b95ffff","8a195da49187fff","8a195da49717fff","8a195da48217fff","8a195da49ca7fff","8a195da494effff","8a195da4d757fff","8a195da48b0ffff","8a195da49857fff","8a195da4b877fff","8a195da4909ffff","8a195da48357fff","8a195da4d0dffff","8a195da4d66ffff","8a195da49407fff","8a195da49997fff","8a195da4d447fff","8a195da48a27fff","8a195da4d21ffff","8a195da4826ffff","8a195da4d7affff","8a195da49547fff","8a194ad32487fff","8a195da48047fff","8a195da48b67fff","8a195da498affff","8a195da490f7fff","8a195da483affff","8a195da49c17fff","8a195da4945ffff","8a195da4d6c7fff","8a195da4d49ffff","8a195da482c7fff","8a195da4900ffff","8a195da49d57fff","8a195da4959ffff","8a195da49907fff","8a195da4d3b7fff","8a195da4b927fff","8a195da4914ffff","8a195da49c6ffff","8a195da4876ffff","8a195da4d71ffff","8a195da494b7fff","8a195da4d4f7fff","8a195da4981ffff","8a195da48ad7fff","8a195da4d2cffff","8a195da49067fff","8a195da4831ffff","8a195da49daffff","8a195da49b87fff","8a195da4d637fff","8a195da4995ffff","8a195da4d40ffff","8a195da491a7fff","8a195da49737fff","8a195da48237fff","8a195da49cc7fff","8a195da4d777fff","8a195da4950ffff","8a195da4800ffff","8a195da49a9ffff","8a195da48b2ffff","8a195da49877fff","8a195da48377fff","8a195da4814ffff","8a195da49427fff","8a195da4d68ffff","8a195da499b7fff","8a195da4d467fff","8a195da4978ffff","8a195da48a47fff","8a195da4828ffff","8a195da49d1ffff","8a195da49567fff","8a195da48067fff","8a195da4bb17fff","8a195da48b87fff","8a195da498cffff","8a195da49117fff","8a195da49c37fff","8a195da4d6e7fff","8a195da48a9ffff","8a195da4b807fff","8a195da4d297fff","8a195da4902ffff","8a195da482e7fff","8a195da49d77fff","8a195da4864ffff","8a195da4bb6ffff","8a195da49397fff","8a195da49927fff","8a195da4b947fff","8a195da4916ffff","8a195da49c8ffff","8a195da494d7fff","8a195da48af7fff","8a195da4d2effff","8a195da4b85ffff","8a195da49087fff","8a195da488cffff","8a195da4d0c7fff","8a195da49ba7fff","8a195da4d657fff"]

```

**RESULT VIEW**
<img width="549" alt="스크린샷 2022-03-21 오후 5 17 10" src="https://user-images.githubusercontent.com/47946430/159224929-59f90070-52f8-40bc-8430-1772fc51a0f6.png">

## Case2. Using H3-go Library / Resolution 9

**Go Test Code**
```
package polyfill_test

import (
	"testing"

	"github.com/daangn/xlog/log"
	uber_h3 "github.com/uber/h3-go/v3"
)

func TestH3Polyfill(t *testing.T) {
	geoCoords := []uber_h3.GeoCoord{
		{51.51746913670682, -0.1251052178242358},
		{51.51981162007689, -0.1254345584085245},
		{51.522082928275104, -0.1264125733143912},
		{51.52421404875121, -0.1280095460538208},
		{51.52614022848794, -0.1301769534182327},
		{51.52780294148869, -0.1328489398316205},
		{51.52915166706029, -0.1359443183389036},
		{51.53014542485875, -0.1393690374304754},
		{51.53075402005622, -0.1430190387497298},
		{51.5309589607956, -0.1467834188535714},
		{51.53075402005622, -0.1505477989574129},
		{51.53014542485875, -0.1541978002766674},
		{51.52915166706029, -0.1576225193682391},
		{51.52780294148869, -0.1607178978755222},
		{51.52614022848794, -0.16338988428891},
		{51.52421404875121, -0.1655572916533219},
		{51.522082928275104, -0.1671542643927515},
		{51.51981162007689, -0.1681322792986182},
		{51.51746913670682, -0.1684616198829069},
		{51.51512665333676, -0.1681322792986182},
		{51.51285534513854, -0.1671542643927515},
		{51.51072422466243, -0.1655572916533219},
		{51.508798044925705, -0.16338988428891},
		{51.50713533192496, -0.1607178978755222},
		{51.505786606353354, -0.1576225193682392},
		{51.50479284855489, -0.1541978002766674},
		{51.50418425335742, -0.1505477989574129},
		{51.50397931261804, -0.1467834188535714},
		{51.50418425335742, -0.1430190387497299},
		{51.50479284855489, -0.1393690374304753},
		{51.505786606353354, -0.1359443183389036},
		{51.50713533192496, -0.1328489398316205},
		{51.508798044925705, -0.1301769534182327},
		{51.51072422466243, -0.1280095460538208},
		{51.51285534513854, -0.1264125733143912},
		{51.51512665333676, -0.1254345584085245},
		{51.51746913670682, -0.1251052178242358},
	}

	const correctH3ResultCnt = 77
	t.Run("h3 polyfill", func(t *testing.T) {
		geoPolygon := uber_h3.GeoPolygon{Geofence: geoCoords}
		h3Idxes := uber_h3.Polyfill(geoPolygon, 9)
		log.Print(h3Idxes)
		if len(h3Idxes) != correctH3ResultCnt {
			t.Errorf("h3 index count is wrong - (correct result: 77, test result: %v)", len(h3Idxes))
		}
	})

}

```

**Result**
```
{"level":"debug","time":"2022-03-21T17:37:20+09:00","message":"[617439388697886719 617439388691857407 617439388700508159 617439388673769471 617439388694478847 617439388688449535 617439388756606975 617439388729868287 617439388670361599 617439388691070975 617439388699721727 617439388672983039 617439388687663103 617439388761849855 617439388681633791 617439388729081855 617439388696313855 617439388758441983 617439388698935295 617439388672196607 617439388686876671 617439388761063423 617439388680847359 617439388695527423 617439388757655551 617439388671410175 617439388692119551 617439388760276991 617439388700770303 617439388694740991 617439388756869119 617439388730130431 617439388676653055 617439388691333119 617439388699983871 617439388673245183 617439388687925247 617439388756082687 617439388681895935 617439388696575999 617438095520563199 617439388758704127 617439388731965439 617439388672458751 617439388761325567 617439388687138815 617439388681109503 617439388695789567 617439388677701631 617439388698411007 617439388692381695 617439388760539135 617439388754509823 617439388701032447 617438095521611775 617439388670885887 617439388691595263 617439388700246015 617439388673507327 617439388694216703 617439388688187391 617439388682158079 617439388696838143 617439388758966271 617439388732227583 617439388672720895 617439388761587711 617439388687400959 617439388681371647 617439388728819711 617439388696051711 617439388692643839 617439388760801279 617439388754771967 617439388680585215 617439388695265279]"}
--- FAIL: TestH3Polyfill (0.00s)
    --- FAIL: TestH3Polyfill/h3_polyfill (0.00s)
        polyfill_test.go:57: h3 index count is wrong - (correct result: 77, test result: 76)
FAIL
exit status 1

```

**Result View**
<img width="593" alt="스크린샷 2022-03-21 오후 5 38 37" src="https://user-images.githubusercontent.com/47946430/159227766-e9b4f9c7-9602-4a4e-9751-411b71969a7d.png">

## Case3. Using h3-py / Resolution 9
**Test Code**
```
import h3
import json

geojson = json.loads('''{"type":"Polygon","coordinates":[[[-0.1251052178242358,51.51746913670682],[-0.1254345584085245,51.51981162007689],[-0.1264125733143912,51.522082928275104],[-0.1280095460538208,51.52421404875121],[-0.1301769534182327,51.52614022848794],[-0.1328489398316205,51.52780294148869],[-0.1359443183389036,51.52915166706029],[-0.1393690374304754,51.53014542485875],[-0.1430190387497298,51.53075402005622],[-0.1467834188535714,51.5309589607956],[-0.1505477989574129,51.53075402005622],[-0.1541978002766674,51.53014542485875],[-0.1576225193682391,51.52915166706029],[-0.1607178978755222,51.52780294148869],[-0.16338988428891,51.52614022848794],[-0.1655572916533219,51.52421404875121],[-0.1671542643927515,51.522082928275104],[-0.1681322792986182,51.51981162007689],[-0.1684616198829069,51.51746913670682],[-0.1681322792986182,51.51512665333676],[-0.1671542643927515,51.51285534513854],[-0.1655572916533219,51.51072422466243],[-0.16338988428891,51.508798044925705],[-0.1607178978755222,51.50713533192496],[-0.1576225193682392,51.505786606353354],[-0.1541978002766674,51.50479284855489],[-0.1505477989574129,51.50418425335742],[-0.1467834188535714,51.50397931261804],[-0.1430190387497299,51.50418425335742],[-0.1393690374304753,51.50479284855489],[-0.1359443183389036,51.505786606353354],[-0.1328489398316205,51.50713533192496],[-0.1301769534182327,51.508798044925705],[-0.1280095460538208,51.51072422466243],[-0.1264125733143912,51.51285534513854],[-0.1254345584085245,51.51512665333676],[-0.1251052178242358,51.51746913670682]]]}''')

h3.polyfill(geojson, 9, geo_json_conformant=True)
```

**RESULT**
```
{'89195da4d3bffff', '89195da482bffff', '89195da4913ffff', '89195da4807ffff', '89195da491bffff', '89195da49cfffff', '89194ad325bffff', '89195da4957ffff', '89195da4b93ffff', '89195da480fffff', '89195da4d77ffff', '89195da4d0bffff', '89195da4827ffff', '89195da4977ffff', '89195da48abffff', '89195da4d6bffff', '89195da49c7ffff', '89195da4993ffff', '89195da49abffff', '89195da4947ffff', '89195da4d2fffff', '89195da48b7ffff', '89195da4983ffff', '89195da4817ffff', '89195da498fffff', '89195da4d7bffff', '89195da4973ffff', '89194ad324bffff', '89195da49bbffff', '89195da4d67ffff', '89195da4917ffff', '89195da4bb7ffff', '89195da483bffff', '89195da490fffff', '89195da4997ffff', '89195da497bffff', '89195da4953ffff', '89195da4d6fffff', '89195da48a7ffff', '89195da4877ffff', '89195da498bffff', '89195da48b3ffff', '89195da4d63ffff', '89195da4903ffff', '89195da48a3ffff', '89195da49dbffff', '89195da4837ffff', '89195da49d3ffff', '89195da4b87ffff', '89195da4867ffff', '89195da490bffff', '89195da4907ffff', '89195da482fffff', '89195da49cbffff', '89195da4d4bffff', '89195da4b97ffff', '89195da495bffff', '89195da4d47ffff', '89195da4d2bffff', '89195da494bffff', '89195da494fffff', '89195da48bbffff', '89195da49d7ffff', '89195da4d0fffff', '89195da4bb3ffff', '89195da499bffff', '89195da4d73ffff', '89195da4d23ffff', '89195da48afffff', '89195da49b3ffff', '89195da4833ffff', '89195da4d4fffff', '89195da4823ffff', '89195da4987ffff', '89195da4b83ffff', '89195da4943ffff'}
```
> Same as case 2 result, the number of h3 cells is 76.

**RESULT VIEW**
<img width="606" alt="스크린샷 2022-03-21 오후 6 07 03" src="https://user-images.githubusercontent.com/47946430/159231584-5316d29a-2e7f-44c2-924f-1c6c9acbc1f6.png">


## Original GeoJSON
Please refer to the original geojson data below.
**Geojson**
```
{"type":"Polygon","coordinates":[[[-0.1251052178242358,51.51746913670682],[-0.1254345584085245,51.51981162007689],[-0.1264125733143912,51.522082928275104],[-0.1280095460538208,51.52421404875121],[-0.1301769534182327,51.52614022848794],[-0.1328489398316205,51.52780294148869],[-0.1359443183389036,51.52915166706029],[-0.1393690374304754,51.53014542485875],[-0.1430190387497298,51.53075402005622],[-0.1467834188535714,51.5309589607956],[-0.1505477989574129,51.53075402005622],[-0.1541978002766674,51.53014542485875],[-0.1576225193682391,51.52915166706029],[-0.1607178978755222,51.52780294148869],[-0.16338988428891,51.52614022848794],[-0.1655572916533219,51.52421404875121],[-0.1671542643927515,51.522082928275104],[-0.1681322792986182,51.51981162007689],[-0.1684616198829069,51.51746913670682],[-0.1681322792986182,51.51512665333676],[-0.1671542643927515,51.51285534513854],[-0.1655572916533219,51.51072422466243],[-0.16338988428891,51.508798044925705],[-0.1607178978755222,51.50713533192496],[-0.1576225193682392,51.505786606353354],[-0.1541978002766674,51.50479284855489],[-0.1505477989574129,51.50418425335742],[-0.1467834188535714,51.50397931261804],[-0.1430190387497299,51.50418425335742],[-0.1393690374304753,51.50479284855489],[-0.1359443183389036,51.505786606353354],[-0.1328489398316205,51.50713533192496],[-0.1301769534182327,51.508798044925705],[-0.1280095460538208,51.51072422466243],[-0.1264125733143912,51.51285534513854],[-0.1254345584085245,51.51512665333676],[-0.1251052178242358,51.51746913670682]]]}
```

**Geojson viewer**
<img width="586" alt="스크린샷 2022-03-21 오후 5 33 32" src="https://user-images.githubusercontent.com/47946430/159227151-130d8a41-0b9f-4688-bc73-6edef57fdf52.png">

## Questions
When I tested various cases using js, python, and go, I couldn't find the exact bug.

In some cases, when polyfilling with resolution 9, h3 cells of the center point were missing, and in another case, when polyfilling with resolution 10, the center point was missing.

Please check if there are any bugs or if my tests are wrong.

--- a/src/apps/testapps/testPolygon.c
+++ b/src/apps/testapps/testPolygon.c
@@ -48,26 +48,82 @@ SUITE(polygon) {
         BBox bbox;
         bboxFromGeoLoop(&geoloop, &bbox);
 
-        // For exact points on the polygon, we bias west and south,
-        // so only the southwest corner is considered in the polygon
+        // For exact points on the polygon, we bias west and south
         t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[0]),
-                 "!contains exact 0");
-        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[1]),
-                 "!contains exact 1");
-        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[2]),
-                 "!contains exact 2");
-        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[4]),
-                 "!contains exact 4");
-        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[5]),
-                 "!contains exact 5");
-
+                 "does not contain exact vertex 0");
         t_assert(pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[3]),
-                 "contains exact 3 (southwest corner)");
+                 "contains exact vertex 3");
 
         t_assert(pointInsideGeoLoop(&geoloop, &bbox, &inside),
                  "contains point inside");
         t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &somewhere),
                  "contains somewhere else");
+    }
+
+    TEST(pointInsideGeoLoopCornerCases) {
+        LatLng verts[] = {{0, 0}, {1, 0}, {1, 1}, {0, 1}};
+        GeoLoop geoloop = {.numVerts = 4, .verts = verts};
+
+        BBox bbox;
+        bboxFromGeoLoop(&geoloop, &bbox);
+
+        LatLng point = {0, 0};
+
+        // Test corners. For exact points on the polygon, we bias west and
+        // north, so only the southeast corner is contained.
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "does not contain sw corner");
+        point.lat = 1;
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "does not contain nw corner ");
+        point.lng = 1;
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "does not contain ne corner ");
+        point.lat = 0;
+        t_assert(pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "contains se corner ");
+    }
+
+    TEST(pointInsideGeoLoopEdgeCases) {
+        LatLng verts[] = {{0, 0}, {1, 0}, {1, 1}, {0, 1}};
+        GeoLoop geoloop = {.numVerts = 4, .verts = verts};
+
+        BBox bbox;
+        bboxFromGeoLoop(&geoloop, &bbox);
+
+        LatLng point;
+
+        // Test edges. Only points on south and east edges are contained.
+        point.lat = 0.5;
+        point.lng = 0;
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "does not contain point on west edge");
+        point.lat = 1;
+        point.lng = 0.5;
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "does not contain point on north edge");
+        point.lat = 0.5;
+        point.lng = 1;
+        t_assert(pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "contains point on east edge");
+        point.lat = 0;
+        point.lng = 0.5;
+        t_assert(pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "contains point on south edge");
+    }
+
+    TEST(pointInsideGeoLoopExtraEdgeCase) {
+        // This is a carefully crafted shape + point to hit an otherwise
+        // missed branch in coverage
+        LatLng verts[] = {{0, 0}, {1, 0.5}, {0, 1}};
+        GeoLoop geoloop = {.numVerts = 4, .verts = verts};
+
+        BBox bbox;
+        bboxFromGeoLoop(&geoloop, &bbox);
+
+        LatLng point = {0.5, 0.5};
+        t_assert(pointInsideGeoLoop(&geoloop, &bbox, &point),
+                 "contains inside point matching longitude of a vertex");
     }
 
     TEST(pointInsideGeoLoopTransmeridian) {

--- a/src/apps/testapps/testPolygon.c
+++ b/src/apps/testapps/testPolygon.c
@@ -48,10 +48,22 @@ SUITE(polygon) {
         BBox bbox;
         bboxFromGeoLoop(&geoloop, &bbox);
 
+        // For exact points on the polygon, we bias west and south,
+        // so only the southwest corner is considered in the polygon
         t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[0]),
-                 "contains exact");
-        t_assert(pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[4]),
-                 "contains exact 4");
+                 "!contains exact 0");
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[1]),
+                 "!contains exact 1");
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[2]),
+                 "!contains exact 2");
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[4]),
+                 "!contains exact 4");
+        t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[5]),
+                 "!contains exact 5");
+
+        t_assert(pointInsideGeoLoop(&geoloop, &bbox, &sfVerts[3]),
+                 "contains exact 3 (southwest corner)");
+
         t_assert(pointInsideGeoLoop(&geoloop, &bbox, &inside),
                  "contains point inside");
         t_assert(!pointInsideGeoLoop(&geoloop, &bbox, &somewhere),

--- a/src/apps/testapps/testPolygonToCellsReported.c
+++ b/src/apps/testapps/testPolygonToCellsReported.c
@@ -170,4 +170,37 @@ SUITE(polygonToCells_reported) {
         t_assert(actualNumIndexes == 4353, "got expected polygonToCells size");
         free(hexagons);
     }
+
+    // https://github.com/uber/h3/issues/595
+    TEST(h3_136) {
+        H3Index center = 0x85283473fffffff;
+        LatLng centerLatLng;
+        H3_EXPORT(cellToLatLng)(center, &centerLatLng);
+
+        // This polygon should include the center cell. The issue here arises
+        // when one of the polygon vertexes is to the east of the index center,
+        // with exactly the same latitude
+        LatLng testVerts[] = {{.lat = centerLatLng.lat, -2.121207808248113},
+                              {0.6565301558937859, -2.1281107217935986},
+                              {0.6515463604919347, -2.1345342663428695},
+                              {0.6466583305904194, -2.1276313527973842}};
+
+        GeoLoop testGeoLoop = {.numVerts = 4, .verts = testVerts};
+        GeoPolygon testPolygon;
+        testPolygon.geoloop = testGeoLoop;
+        testPolygon.numHoles = 0;
+
+        int res = 5;
+        int64_t numHexagons;
+        t_assertSuccess(H3_EXPORT(maxPolygonToCellsSize)(&testPolygon, res, 0,
+                                                         &numHexagons));
+        H3Index *hexagons = calloc(numHexagons, sizeof(H3Index));
+
+        t_assertSuccess(
+            H3_EXPORT(polygonToCells)(&testPolygon, res, 0, hexagons));
+        int64_t actualNumIndexes = countNonNullIndexes(hexagons, numHexagons);
+
+        t_assert(actualNumIndexes == 8, "got expected polygonToCells size");
+        free(hexagons);
+    }
 }

--- a/src/h3lib/include/polygonAlgos.h
+++ b/src/h3lib/include/polygonAlgos.h
@@ -94,7 +94,14 @@ bool GENERIC_LOOP_ALGO(pointInside)(const TYPE *loop, const BBox *bbox,
 
         // If the latitude matches exactly, we'll hit an edge case where
         // the ray passes through the vertex twice on successive segment
-        // checks. To avoid this, adjust the latiude northward if needed
+        // checks. To avoid this, adjust the latiude northward if needed.
+        //
+        // NOTE: This currently means that a point at the north pole cannot
+        // be contained in any polygon. This is acceptable in current usage,
+        // because the point we test in this function at present is always
+        // a cell center or vertex, and no cell has a center or vertex on the
+        // north pole. If we need to expand this algo to more generic uses we
+        // might need to handle this edge case.
         if (lat == a.lat || lat == b.lat) {
             lat += DBL_EPSILON;
         }

--- a/src/h3lib/include/polygonAlgos.h
+++ b/src/h3lib/include/polygonAlgos.h
@@ -92,6 +92,13 @@ bool GENERIC_LOOP_ALGO(pointInside)(const TYPE *loop, const BBox *bbox,
             b = tmp;
         }
 
+        // If the latitude matches exactly, we'll hit an edge case where
+        // the ray passes through the vertex twice on successive segment
+        // checks. To avoid this, adjust the latiude northward if needed
+        if (lat == a.lat || lat == b.lat) {
+            lat += DBL_EPSILON;
+        }
+
         // If we're totally above or below the latitude ranges, the test
         // ray cannot intersect the line segment, so let's move on
         if (lat < a.lat || lat > b.lat) {

